jsparagus/grammar.py

```diff
@@ -117,16 +117,29 @@ def copy_with(self, **kwargs):
 # used only in productions for init nonterminals, created automatically by
 # Grammar.__init__(). It's not a reduce expression, so it can't be nested.
 #
-CallMethod = collections.namedtuple("CallMethod", "method args")
+
+class CallMethod(collections.namedtuple("CallMethod", "method args trait")):
+    """Express a method call, and give it a given set of arguments. A trait is
+    added as the parser should implement this trait to call this method."""
+    def __new__(cls, method, args, trait):
```

Comment:
might be nice to add default value for `trait`, instead of repeating `"AstBuilder"` in many callers